pkg/controller/node/node_controller.go

```diff
@@ -1059,6 +1059,8 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 				}
 			}
 		}
+
+		time.Sleep(30 * time.Second)
```

Comment:
A few notes:
1. I think based on a brief search of how a connection reset can happen, the most likely possibility is that we're sending too many requests? Although there is a 20s delay in the test script (if I'm reading that correctly) so I'm not sure if we're just getting unlucky
2. 30 seconds is definitely too long. We have a updateDelay built into the controller which defaults to 5s, we do use it in one place in the sync loop, and we can consider expanding that to cover more case that return to see if that helps
3. The worker and master pool sync loops are separate, that said I think masters and workers will update in parallel since we have a worker count of `2` for each of the subcontrollers, so maybe the parallel requests is what's causing the inter-pool weirdness?

Reply:
Sorry, looking at the original error in the bug, it actually seems to be coming from us trying to set the working status in the daemon here: https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/update.go#L993

I think a good thing to check when reproducing is: is it always the setworking() that's erroring, or are there other conditions in the daemon that can fail.

Also, while slowing down updates in the controller might help, I'm not sure if the issue is directly tackled. I think it might be best to instead have the daemon status setting have a retry (something like 3 retries with 5s backoff might be enough) to get rid of api blips